fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneAutoCompleteResultCursor.java

```diff
@@ -0,0 +1,183 @@
+            if (lookupResults == null) {
+                try {
+                    performLookup();
+                } catch (IOException ioException) {
+                    throw new RecordCoreException("Exception to lookup the auto complete suggestions", ioException)
+                            .addLogInfo(LogMessageKeys.QUERY, query);
+                }
+            }
+            return currentPosition < lookupResults.size() ? lookupResults.get(currentPosition) : null;
+        }, executor);
+
+        return lookupResult.thenApply(r -> {
+            if (r == null) {
+                return RecordCursorResult.exhausted();
+            } else {
+                final String suggestion = highlight ? (String) r.highlightKey : (String) r.key;
+
+                if (r.payload == null) {
+                    throw new RecordCoreException("Empty payload of lookup result for lucene auto complete suggestion")
+                            .addLogInfo(LogMessageKeys.QUERY, query)
+                            .addLogInfo(LogMessageKeys.RESULT, suggestion);
+                }
+
+                IndexEntry indexEntry = new IndexEntry(state.index,
+                        Tuple.fromBytes(r.payload.bytes).add(suggestion),
+                        Tuple.from(r.value));
+
+                if (LOGGER.isTraceEnabled()) {
+                    LOGGER.trace("Suggestion read as an index entry={}", indexEntry);
+                }
+
+                return RecordCursorResult.withNextValue(indexEntry, continuationHelper(lookupResults.get(currentPosition++)));
+            }
+        });
+    }
+
+    @Nonnull
+    private static RecordCursorContinuation continuationHelper(@Nonnull Lookup.LookupResult lookupResult) {
+        RecordCursorProto.LuceneAutoCompleteIndexContinuation.Builder continuationBuilder = RecordCursorProto.LuceneAutoCompleteIndexContinuation.newBuilder().setKey((String) lookupResult.key);
+        continuationBuilder.setValue(lookupResult.value);
+        continuationBuilder.setPayload(ByteString.copyFrom(lookupResult.payload.bytes));
+        return ByteArrayContinuation.fromNullable(continuationBuilder.build().toByteArray());
+    }
+
+    @Override
+    public void close() {
+        IOUtils.closeWhileHandlingException(suggester);
+    }
+
+    @Nonnull
+    @Override
+    public Executor getExecutor() {
+        return executor;
+    }
+
+    @Override
+    public boolean accept(@Nonnull RecordCursorVisitor visitor) {
+        visitor.visitEnter(this);
+        return visitor.visitLeave(this);
+    }
+
+    private void performLookup() throws IOException {
+        if (lookupResults != null) {
+            return;
+        }
+        long startTime = System.nanoTime();
+        lookupResults = suggester.lookup(searchTerm,
+                fieldName == null ? Collections.emptySet() : Collections.singleton(new BytesRef(fieldName.getBytes(StandardCharsets.UTF_8))),
```

Comment:
does this default with a null field name to the suggester checking across all fields?

Reply:
additionally its possible that we could pull the field names and pass them through the constructor. That would avoid us mistakenly checking a field that we're not intending to such as the primary key or maybe any other fields we make in the future.

Reply:
Added check for the query to be parsed for search term and field, which is pretty straightforward. please let me know if it looks good to you

Reply:
does this work in terms of checking when a field isn't provided? I guess this isn't an issue for now because we pass fields via the lucene query function but somethingtothink about for later.